### DISCUSSION TITLE
Certain Windows Event logs not shipped by Winlogbeat

### DISCUSSION THREAD

**Discussion by Rdago** [Security-Onion-Solutions/securityonion#6907] [Unsupported Versions] [answered]
Hi there, 

we are currently facing some issues with winlogbeat not shipping certain security-eventlogs. We noticed that for example event id 4732 is not getting shipped. Event ID 4627, which also comed from the security event log is getting shipped flawless.

### logstash analysis

We tried to forward the events to a local log file from logstash in order to debug the issue (https://docs.securityonion.net/en/latest/logstash.html#forwarding-events-to-an-external-destination) and couldn' t see this specific event log. 

### Winlogbeat analysis

tried winlogbeat versions: 
- 7.16.3
- beta 8.0.0
- 6.8.22
- 7.16.2 (.msi shipped with SO)

winlogbeat.yml : 
```
winlogbeat.event_logs:
  - name: Security

setup.template.settings:
  index.number_of_shards: 1

output.logstash:
  hosts: ["<so-instance>:5044"]
```

Winlogbeat Debug Logging:

We noticed that the events are getting logged in the Event Viewer and winlogbeat is trying to ship them: 
`2022-01-18T16:56:06.676+0100	DEBUG	[processors]	processing/processors.go:203	Publish event: { <The Event 4732> [...]`

Sometimes it shows `RenderErrorDataItemName` errors regarding this event id, but it does on others which do appear in logstash too: 

```
[...]
Restricted SID Count:	0 Level:Information Task:SAM Opcode:Info Keywords:[Audit Success] RenderErrorCode:0 RenderErrorDataItemName: RenderErr:[]}
2022-01-18T16:56:06.672+0100	DEBUG	[eventlog_detail]	eventlog/wineventlog.go:350	WinEventLog[Security] XML=<Event xmlns= 
[...]

Additional Information:
	Privileges:		- Level:Information Task:Security Group Management Opcode:Info Keywords:[Audit Success] RenderErrorCode:0 RenderErrorDataItemName: RenderErr:[]}

[...]
```

**Comment by Rdago** [comment, marked as answer]
Suggestion is to adjust the beat field limit to 2000 ( "index.mapping.total_fields.limit": 2000)
Field Limit was the Problem: ---> https://docs.securityonion.net/en/2.3/elasticsearch.html#field-limit

**Reply by defensivedepth** [reply]
@Rdago Thanks for posting the solution.

How did you come to this conclusion?

**Reply by Git-Me-Some-Hub** [reply]
@Rdago , I am also very intrigued to know about the fix for this. I am seeing this sporadically as well.

**Reply by Rdago** [reply]
From `/opt/so/log/logstash/logstash-2022-01-19.log`
```
[2022-01-19T00:29:33,748][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"so-beats", :routing=>nil, :pipeline=>"beats.common"}, {"message"=>" REDACTED "version"=>"7.16.3", "type"=>"winlogbeat", "ephemeral_id"=>"b355c810-6ebc-4e56-b005-a05c4012cfd5"}, "error"=>{"message"=>"XML syntax error on line 41: invalid character entity & (no semicolon)"}, "tags"=>["beat-ext", "beats_input_codec_plain_applied"], "type"=>"redis-input"}], :response=>{"index"=>{"_index"=>"so-beats-2022.01.19", "_type"=>"_doc", "_id"=>"REDACTED", "status"=>400, "error"=>{"type"=>"mapper_parsing_exception", "reason"=>"failed to parse", "caused_by"=>{"type"=>"illegal_argument_exception", "reason"=>"Limit of total fields [1500] has been exceeded while adding new fields [1]"}}}}}
```

**Reply by Git-Me-Some-Hub** [reply]
Thanks @Rdago , I do know that I have these same filed limit errors and have for some time. The reason I have not changed them is more around me not totally understanding the process to update them. Do I need to do an elastic query to update the settings or is it in a config file for SALT somewhere. I found it very unclear in the documentation. 

Much appreciated.

**Reply by dougburks** [reply]
The documentation shows a temporary change (https://docs.securityonion.net/en/2.3/elasticsearch.html#temporary) and a persistent change (https://docs.securityonion.net/en/2.3/elasticsearch.html#persistent). From the persistent section, you'll most likely want both:
> Please note that the change to the field limit will not occur immediately, only on index creation. Therefore, it is recommended to run the previously mentioned temporary command and modify the template file.
